src/app/services/xml-parsers/bibliography-parser.service.ts

```diff
@@ -0,0 +1,147 @@
+              publishers,
+              biblScopes,
+              notes,
+              series,
+            };
+            if (!bibliographicCitations.includes(biblCitation)) {
+              bibliographicCitations.push(biblCitation);
+              this.findedCits.complete++;
+              this.findedCits.total++;
+            }
+          }
+        });
+
+        Array.from(responses.getElementsByTagName('biblStruct')).map(citation => {
+          const biblStructCitation: BibliographicCitation = {
+            analytic_authors: '',
+            analytic_titles: '',
+            monogr_authors: '',
+            monogr_titles: '',
+            monogr_editors: '',
+            monogr_imprint_pubPlaces: '',
+            monogr_imprint_publishers: '',
+            monogr_imprint_dates: '',
+            monogr_biblScopes: '',
+            series_titles: '',
+            series_biblScopes: '',
+            notes: '',
+          };
+          Array.from(citation.getElementsByTagName('analytic')).map(features => {
+            biblStructCitation.analytic_authors += ' ' + getStringFromTags('author', features);
+            biblStructCitation.analytic_titles += ' ' + getStringFromTags('title', features);
+          });
+          Array.from(citation.getElementsByTagName('monogr')).map(features => {
+            biblStructCitation.monogr_authors += ' ' + getStringFromTags('author', features);
+            biblStructCitation.monogr_titles += ' ' + getStringFromTags('title', features);
+            biblStructCitation.monogr_editors += ' ' + getStringFromTags('editor', features);
+            Array.from(features.getElementsByTagName('imprint')).map(subFeatures => {
+              biblStructCitation.monogr_imprint_pubPlaces += ' ' + getStringFromTags('pubPlace', subFeatures);
+              biblStructCitation.monogr_imprint_publishers += ' ' + getStringFromTags('publisher', subFeatures);
+              biblStructCitation.monogr_imprint_dates += ' ' + getStringFromTags('date', subFeatures);
+            });
+            biblStructCitation.monogr_biblScopes += ' ' + getStringFromTags('biblScope', features);
+          });
+          Array.from(citation.getElementsByTagName('series')).map(features => {
+            biblStructCitation.series_titles += ' ' + getStringFromTags('title', features);
+            biblStructCitation.series_biblScopes += ' ' + getStringFromTags('biblScope', features);
+          });
+          biblStructCitation.notes += ' ' + getStringFromTags('note', citation);
+          if (!Object.values(biblStructCitation).every(x => (x === '')) &&
+            !bibliographicCitations.includes(biblStructCitation)) {
+            bibliographicCitations.push(biblStructCitation);
+            this.findedCits.complete++;
+            this.findedCits.total++;
+          }
+        });
+
+        return { citations: bibliographicCitations, areComplete: ((this.findedCits.complete === this.findedCits.total) ? true : false) };
```

Comment:
Just use the condition for `areComplete` value.
Change `findedCits` to `foundCits`

What does it mean that `areComplete` is true when `complete` e `total` are equal?

Reply:
Please use meaningful and understandable variable names